src/http/HttpHeaderBag.php

```diff
@@ -37,28 +59,97 @@ public function get($key) {
     if (array_key_exists($key, $this->values)) {
       return $this->values[$key];
     }
-    return NULL;
+    // Return a NULL object on which you can still call the HttpHeaderInterface
+    // methods.
+    return HttpHeaderNull::create(NULL, NULL);
```

Comment:
The problem of that is that it muddies the interface by having:

```php
/**
  * @return string|HttpHeaderInterface
  */
```

Which means that your method chaining is cut depending on the passed in arguments. I'd rather have methods returning a single interface or `NULL`.